workflow-job

```diff
@@ -215,14 +224,32 @@ def isAlive(dockerContainer) {
    }
 }
 
+def waitGuiceServerAndRunTestOnThat(dockerContainer, jamesCliCmd) {
+    echo 'Debug step 1'
+    echo "${dockerContainer}"
+    echo 'Debug step 2'
+    echo "${jamesCliCmd}"
+    echo 'Waiting for James server to be deployed'
+    try {
+        waitForCommandSuccess("docker exec ${dockerContainer} ${jamesCliCmd} listusers", maxRetries)
+    } catch (Exception e) {
+        sh "docker logs ${dockerContainer}"
+    }
+    deploying.success();
+
+    sh "docker exec ${dockerContainer} ${jamesCliCmd} adddomain domain"
+    sh "docker exec ${dockerContainer} ${jamesCliCmd} adduser imapuser@domain password"
+}
```

Comment:
You are not plyaing integration tests, missing:
sh "${findImapPort} && docker run ${verbose} --name=${containers.integration} ...

command.

It's a good idea to factorize things between Cassandra/JPA, you should remove the stage "Configuring James Server" which is only used for Cassandra integration tests part.